codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AssembleMiddlewareStack.java

```diff
@@ -187,4 +199,47 @@ public List<RuntimeClientPlugin> getClientPlugins() {
                         .build()
         );
     }
+
+    @Override
+    public void writeAdditionalFiles(GoSettings settings, Model model, SymbolProvider symbolProvider, GoDelegator goDelegator) {
+        goDelegator.useFileWriter("api_client.go", settings.getModuleName(), addMiddleware());
+    }
+
+    // TODO symbols
```

Comment:
The commit two entries down addresses this: https://github.com/aws/aws-sdk-go-v2/pull/2516/commits/0d1b977e61194563ccf40ed58af31499d8d6681b